docs/guides/nextauth-migration.mdx

```diff
@@ -0,0 +1,362 @@
+- Protect your app
+- Read user and session data
+- User IDs as Foreign Keys
+- Create a production instance
+- Migrate users to a production instance
+- Further support
+
+</TutorialHero>
+
+This guide shows how to migrate an application using NextAuth.js to use Clerk for authentication.
+
+<Steps>
+
+### Install `@clerk/nextjs`
+
+Clerk's Next.js SDK gives you access to prebuilt [components](https://clerk.com/docs/components/overview), [hooks](https://clerk.com/docs/references/nextjs/overview#client-side-helpers), and [helpers](https://clerk.com/docs/references/nextjs/overview) for Next.js Server Components, Route Handlers and Middleware. Install it by running the following command in your terminal:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+  ```bash filename="terminal"
+  npm install @clerk/nextjs
+  ```
+
+```bash filename="terminal"
+yarn add @clerk/nextjs
+```
+
+```bash filename="terminal"
+pnpm add @clerk/nextjs
+```
+
+</CodeBlockTabs>
+
+### Set environment keys
+
+Store the keys in the `.env.local` file of your Next.js application.
+
+<InjectKeys>
+
+```sh filename=".env.local"
+NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
+CLERK_SECRET_KEY={{secret}}
+```
+
+</InjectKeys>
+
+### Wrap your Next.js app in `<ClerkProvider />`
+
+#### Remove your NextAuth.js `<SessionProvider />`
+
+Replace `<SessionProvider session={session}>` with `<ClerkProvider>` if you are using App Router or `<ClerkProvider {...pageProps}>` if you are using Pages Router.
+
+```tsx filename="_app.tsx"
+// With NextAuth.js
+
+import { SessionProvider } from "next-auth/react"
+
+export default function App({
+  Component,
+  pageProps: { session, ...pageProps },
+}) {
+  return (
+    <SessionProvider session={session}>
+      <Component {...pageProps} />
+    </SessionProvider>
+  )
+}
+```
+
+#### Add `<ClerkProvider />` to your root `layout.tsx` file
+
+<CodeBlockTabs type="installer" options={["App Router", "Pages Router"]}>
+
+```tsx filename="app/layout.tsx"
+import { ClerkProvider } from '@clerk/nextjs'
+import './globals.css'
+ 
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <ClerkProvider>
+      <html lang="en">
+        <body>{children}</body>
+      </html>
+    </ClerkProvider>
+  )
+}
+```
+
+```tsx filename="_app.tsx"
+import '@/styles/globals.css'
+import { ClerkProvider } from "@clerk/nextjs";
+import type { AppProps } from "next/app";
+function MyApp({ Component, pageProps }: AppProps) {
+  return (
+    <ClerkProvider {...pageProps}>
+      <Component {...pageProps} />
+    </ClerkProvider>
+  );
+}
+export default MyApp;
+```
+</CodeBlockTabs>
+
+### Setup Sign-up and sign-in UI
+
+#### Account Portal
+
+Account Portal is the fastest way to authenticate your app. Clerk's Account Portal hosts the `<SignIn />`, `<SignUp />`, `<UserProfile />` and other components for your application. Read more about [Account Portal](docs/account-portal/getting-started). 
```

Comment:
```suggestion
Account Portal is the fastest way to authenticate your app. Clerk's Account Portal hosts the `<SignIn />`, `<SignUp />`, `<UserProfile />`, and other components for your application. Read more about [Account Portal](/docs/account-portal/getting-started). 
```

Reply:
lmao oxford comma. 

Is that `/` needed? did the link work before?

Reply:
Yes, the lack of `/` creates http://localhost:4003/docs/docs/account-portal/getting-started -- note the `docs/docs/` in the middle.